src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -439,44 +439,110 @@ Vector128<ulong> SoftwareFallback(ulong x)
         /// <param name="e14">The value that element 14 will be initialized to.</param>
         /// <param name="e15">The value that element 15 will be initialized to.</param>
         /// <returns>A new <see cref="Vector128{Byte}" /> with each element initialized to corresponding specified value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector128<byte> Create(byte e0, byte e1, byte e2, byte e3, byte e4, byte e5, byte e6, byte e7, byte e8, byte e9, byte e10, byte e11, byte e12, byte e13, byte e14, byte e15)
         {
-            var pResult = stackalloc byte[16]
-            {
-                e0,
-                e1,
-                e2,
-                e3,
-                e4,
-                e5,
-                e6,
-                e7,
-                e8,
-                e9,
-                e10,
-                e11,
-                e12,
-                e13,
-                e14,
-                e15,
-            };
+            if (Sse41.IsSupported)
+            {
+                Vector128<byte> result = CreateScalarUnsafe(e0);                                    // <  0, ??, ??, ??, ??, ??, ??, ??, ??, ??, ??, ??, ??, ??, ??, ?? >
```

Comment:
This is currently set to always inline (which matches what the native `_mm_set_epi8` intrinsic does). The method is otherwise too large/complex for the JIT to consider inlining otherwise.

This looks to generally result  in better codegen when inlined anyways, since the JIT has to do a bunch of loads/stores to make the method call anyways.
-- It will also matter much less once we add support for emitting a memory constant when all or most of the inputs are themselves constant, since that is expected to be the common case anyways

Reply:
Also, for the non-inlined case; the codegen is pretty bad due to the `movzx`/`movsx` the JIT inserts (@CarolEidt mentioned this was due to the `normalizeOnLoad` semantics for parameters; which is changed to `normalizeOnStore` when inlined: https://github.com/dotnet/coreclr/pull/21432#discussion_r239902701)